pkg/controller/job/job_controller_test.go

```diff
@@ -1855,10 +1861,9 @@ func TestSyncPastDeadlineJobFinished(t *testing.T) {
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
 			job := newJobWithName(tc.jobName, 1, 1, 6, batch.NonIndexedCompletion)
-			activeDeadlineSeconds := int64(1)
-			job.Spec.ActiveDeadlineSeconds = &activeDeadlineSeconds
+			job.Spec.ActiveDeadlineSeconds = pointer.Int64(1)
 			if tc.setStartTime {
-				start := metav1.Unix(metav1.Now().Time.Unix()-1, 0)
+				start := metav1.Unix(manager.clock.Now().Unix()-1, 0)
```

Comment:
this looks too cryptic:

```suggestion
				start := metav1.Time(fakeClock.Now().Sub(time.Second))
```

I think the original code was trying to truncate too, but I don't think that's important. But if you really want it, you can change the clock instead:

```golang
fakeClock := clocktesting.NewFakeClock(time.Now().Truncate(time.Second))
```

Reply:
`metav1.Time(fakeClock.Now().Sub(time.Second))` This can't be done right ? The `.Sub` returns a `time.Duration` where as the `metav1.NewTime` needs a `time.Time`

Instead can I just do `metav1.NewTime(clocktesting.NewFakeClock(time.Now().Truncate(time.Second)).Now())` instead ? @alculquicondor

Reply:
oh, I guess it is `time.Add(-time.Second)`.

Reply:
/Ack. Took care of both.